cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -473,10 +526,9 @@ endif()
 
 #---Check for GSL library---------------------------------------------------------------
 if(mathmore OR builtin_gsl)
-  if(builtin_gsl AND NO_CONNECTION)
-    if(fail-on-missing)
-      message(FATAL_ERROR "No internet connection. Please check your connection, or either disable the 'builtin_gsl' option or the 'fail-on-missing' to automatically disable options requiring internet access")
-    else()
+  if(builtin_gsl)
+    ROOT_CHECK_CONNECTION("builtin_gsl=OFF")
+    if(NO_CONNECTION)
       message(STATUS "No internet connection, disabling 'builtin_gsl' option")
       set(builtin_gsl OFF CACHE BOOL "Disabled because there is no internet connection" FORCE)
```

Comment:
It feels that the next 2 lines could also be folded into the `ROOT_CHECK_CONNECTION` but
* They are repeated only 90% (ish) of the times, the other 10% different messages is printed or additional flags are disabled.
* It is helpful to "see" that the option that was just check true (here `builtin_gsl` is turned off).

So maybe we can have a second wrapper to cover those 90%:
```
macro(ROOT_CHECK_CONNECTION_AND_DISABLE_OPTION option)
   ROOT_CHECK_CONNECTION("$(option)=OFF")
   if(NO_CONNECTION)
      message(STATUS "No internet connection, disabling '${option}' option")
      set(${option} OFF CACHE BOOL "Disabled because there is no internet connection" FORCE)
   endif()
endmacro()
```

Reply:
That's a nice idea!